File: posixlib/src/main/resources/scala-native/fnmatch.c
#if defined(SCALANATIVE_COMPILE_ALWAYS) || defined(__SCALANATIVE_POSIX_FNMATCH)
#if defined(__unix__) || defined(__unix) || defined(unix) ||                   \
    (defined(__APPLE__) && defined(__MACH__))

#include <fnmatch.h>

int scalanative_fnm_nomatch() { return FNM_NOMATCH; };

int scalanative_fnm_pathname() { return FNM_PATHNAME; };

int scalanative_fnm_period() { return FNM_PERIOD; };

int scalanative_fnm_noescape() { return FNM_NOESCAPE; };

#endif // Unix or Mac OS
#endif

File: posixlib/src/main/scala/scala/scalanative/posix/fnmatch.scala
package scala.scalanative
package posix

import scalanative.unsafe._

/** POSIX fnmatch.h for Scala
 *
 *  The Open Group Base Specifications
 *  [[https://pubs.opengroup.org/onlinepubs/9699919799 Issue 7, 2018]] edition.
 */

@extern
object fnmatch {
  // Symbolic constants

  @name("scalanative_fnm_nomatch")
  def FNM_NOMATCH: CInt = extern

  @name("scalanative_fnm_pathname")
  def FNM_PATHNAME: CInt = extern

  @name("scalanative_fnm_period")
  def FNM_PERIOD: CInt = extern

  @name("scalanative_fnm_noescape")
  def FNM_NOESCAPE: CInt = extern

  // Method

  def fnmatch(pattern: CString, string: CString, flags: CInt): CInt = extern
}
